File: r/bioviz/tests/testthat/test-igv.R
test_that("igv() assembles a config from convenience args", {
  w <- igv(
    genome = "hg38",
    locus = "chr8:127,736,588-127,739,371",
    tracks = list(list(name = "t", url = "t.bw", format = "bigWig"))
  )

  expect_s3_class(w, "htmlwidget")
  # Config-driven: no data columns.
  expect_equal(w$x$data$columns, list())
  expect_equal(w$x$options$genome, "hg38")
  expect_equal(w$x$options$locus, "chr8:127,736,588-127,739,371")
  expect_equal(w$x$options$tracks[[1]]$url, "t.bw")
  # No explicit config passed through.
  expect_null(w$x$options$config)
})

test_that("igv() passes a full config through and ignores convenience args", {
  cfg <- list(genome = "hg19", locus = "chr1:1-1000")
  w <- igv(genome = "hg38", locus = "chr2", config = cfg)

  expect_equal(w$x$options$config, cfg)
  expect_null(w$x$options$genome)
  expect_null(w$x$options$locus)
})

test_that("igv() omits tracks when none are supplied", {
  w <- igv(genome = "hg38")
  expect_equal(w$x$options$genome, "hg38")
  expect_null(w$x$options$tracks)
})

test_that("igv() validates its input", {
  expect_error(igv(config = "not-a-list"), "must be a named list")
  expect_error(igv(tracks = "not-a-list"), "must be a list")
})

File: pkg-py/tests/test_igv.py
from plotomics import IGV


def test_igv_assembles_config_from_convenience_args():
    w = IGV(
        genome="hg38",
        locus="chr8:127,736,588-127,739,371",
        tracks=[{"name": "t", "url": "t.bw", "format": "bigWig"}],
    )
    assert w.options["genome"] == "hg38"
    assert w.options["locus"] == "chr8:127,736,588-127,739,371"
    assert w.options["tracks"][0]["url"] == "t.bw"
    assert "config" not in w.options
    # Config-driven: no data columns.
    assert w.data["columns"] == {}


def test_igv_passes_full_config_through():
    cfg = {"genome": "hg19", "locus": "chr1:1-1000"}
    w = IGV(genome="hg38", locus="chr2", config=cfg)
    assert w.options["config"] == cfg
    assert "genome" not in w.options
    assert "locus" not in w.options


def test_igv_omits_tracks_when_none():
    w = IGV(genome="hg38")
    assert w.options["genome"] == "hg38"
    assert "tracks" not in w.options


def test_igv_defaults_to_hg38():
    w = IGV()
    assert w.options["genome"] == "hg38"
    assert w._height == 480
